src/tablestore.rs

```diff
@@ -213,17 +205,19 @@ impl TableStore {
                 _ => SlateDBError::ObjectStoreError(e),
             })?;
 
-        self.cache_filter(*id, encoded_sst.filter);
+        self.cache_filter(*id, encoded_sst.info.filter_offset, encoded_sst.filter)
+            .await;
         Ok(SsTableHandle {
             id: *id,
             info: encoded_sst.info,
         })
     }
 
-    fn cache_filter(&self, sst: SsTableId, filter: Option<Arc<BloomFilter>>) {
-        {
-            let mut wguard = self.filter_cache.write();
-            wguard.insert(sst, filter);
+    async fn cache_filter(&self, sst: SsTableId, id: u64, filter: Option<Arc<BloomFilter>>) {
```

Comment:
looks like filter offset is the id, should the parameter be called `filter_offset`?

Reply:
Also, I see that key for block is `(SsTableId, block_num)` and key for filter is `(SsTableId, filter_offset)`.
Wouldn't there be an issue if `block_num == filter_offset`?

Reply:
> key for block is (SsTableId, block_num) 

Where are you seeing this? The PR used to have this, but it was fixed:

```rust
                    blocks_to_cache.push((handle.id, offset, block.clone()));
```

Reply:
I don't see it anymore.